### PR TITLE
Armor items tweaks

### PR DESCRIPTION
**Pull request by Rail-Runner**
1. Change large suitcase, medium suitcase, swag bag, plastic shopping bag, straw basket, bindle, briefcase and briefcase SMG to cover one of two hands, not both, and double their encumbrance.
2. Add OVERSIZE, BELTED and FANCY flags to briefcase SMG.
3. Add OVERSIZE and BELTED flags to large suitcase, medium suitcase, swag bag, plastic shopping bag and straw basket.
4. Add BELTED flag to briefcase and bindle.
5. Add OVERSIZE flag to shoulder strap, makeshift sling and helmet netting.
6. Add 5 coverage to medium suitcase.
7. Add 10 storage to salvaged power armor. UPS compartment it still going nowhere even if you took out all the electronics.
8. Fix mittens flags.
9. Add COLLAR flag to survivor duster and sleeveless survivor duster.
10. Remove unneded "power_armor" values from salvaged power armor and salvaged power armor helmet.
11. Add ALLOWS_NATURAL_ATTACKS flag to wrist watch, diver's watch, gold watch, fancy watch, game watch, turban, helmet netting and pair of sock mitts.
12. Add VARSIZE flag to chef's jacket and checkered pants.
13. Add POCKETS flag to leather vest, leather body armor, boiled leather armor, plated leather armor, fur body armor, ski pants, fur pants, denim shorts, subject suit, tuxedo and SWAT armor.
14. Remove SKINTIGHT flag from work t-shirts.

### CODE REVIEW

**Comment by Zireael07** [comment]
Why fancy for SMG briefcase? And 18 storage is a bit too much.

File: data/json/items/armor.json [hunk 1]
```diff
@@ -7322,11 +7322,12 @@
         "name" : "suitcase",
         "weight" : 5000,
         "color" : "red",
-        "covers" : ["ARMS", "HANDS", "LEGS"],
+        "covers" : ["ARM_EITHER", "HAND_EITHER", "LEG_EITHER"],
```
Comment (Coolthulhu) [inline data/json/items/armor.json]: This halves the overall encumbrance, you should double the encumbrance to compensate for that.

File: data/json/items/armor.json [hunk 2]
```diff
@@ -7344,11 +7345,12 @@
         "name" : "suitcase",
         "weight" : 900,
         "color" : "blue",
-        "covers" : ["ARMS", "HANDS"],
+        "covers" : ["ARM_EITHER", "HAND_EITHER"],
```
Comment (Coolthulhu) [inline data/json/items/armor.json]: Same here - also needs doubled encumbrance.

File: data/json/items/armor.json [hunk 3]
```diff
@@ -10908,11 +10910,12 @@
         "name" : "swag bag",
         "weight" : 415,
         "color" : "brown",
-        "covers" : ["ARMS", "HANDS"],
+        "covers" : ["ARM_EITHER", "HAND_EITHER"],
```
Comment (Coolthulhu) [inline data/json/items/armor.json]: Also here

File: data/json/items/armor.json [hunk 4]
```diff
@@ -10930,7 +10933,7 @@
         "name" : "briefcase",
         "weight" : 1700,
         "color" : "light_gray",
-        "covers" : ["ARMS", "HANDS"],
+        "covers" : ["ARM_EITHER", "HAND_EITHER"],
```
Comment (Coolthulhu) [inline data/json/items/armor.json]: And here

File: data/json/items/armor.json [hunk 5]
```diff
@@ -10953,7 +10956,7 @@
         "name" : "bindle",
         "weight" : 980,
         "color" : "light_gray",
-        "covers" : ["ARMS", "HANDS"],
+        "covers" : ["ARM_EITHER", "HAND_EITHER"],
```
Comment (Coolthulhu) [inline data/json/items/armor.json]: And here

File: data/json/items/armor.json [hunk 6]
```diff
@@ -10976,11 +10979,12 @@
         "name" : "plastic shopping bag",
         "weight" : 2,
         "color" : "white",
-        "covers" : ["ARMS", "HANDS"],
+        "covers" : ["ARM_EITHER", "HAND_EITHER"],
```
Comment (Coolthulhu) [inline data/json/items/armor.json]: And here

File: data/json/items/armor.json [hunk 7]
```diff
@@ -11855,11 +11859,12 @@
         "name" : "straw basket",
         "weight" : 100,
         "color" : "light_gray",
-        "covers" : ["ARMS", "HANDS"],
+        "covers" : ["ARM_EITHER", "HAND_EITHER"],
```
Comment (Coolthulhu) [inline data/json/items/armor.json]: And the last one

File: data/json/items/ranged.json [hunk 8]
```diff
@@ -382,8 +382,9 @@
     "cutting": 0,
     "to_hit": 0,
     "armor_data" : {
-        "covers" : ["ARMS", "HANDS"],
+        "covers" : ["ARM_EITHER", "HAND_EITHER"],
         "coverage" : 10,
+		"storage" : 18,
```
Comment (Coolthulhu) [inline data/json/items/ranged.json]: Tabs

File: data/json/items/ranged.json [hunk 9]
```diff
@@ -382,8 +382,9 @@
     "cutting": 0,
     "to_hit": 0,
     "armor_data" : {
-        "covers" : ["ARMS", "HANDS"],
+        "covers" : ["ARM_EITHER", "HAND_EITHER"],
```
Comment (Coolthulhu) [inline data/json/items/ranged.json]: OK, this one also needs the encumbrance adjusted

**Comment by Coolthulhu** [comment]
> Add POCKETS flag to leather vest, leather body armor, boiled leather armor, plated leather armor, fur body armor, shorts, basketball shorts, mail carrier shorts, ski pants, fur pants, leather pants, denim shorts, checkered pants, fencing pants, subject suit, tuxedo, mail carrier shirt, chest rig, MBR vest and SWAT armor.

Not too sure about this one.
Some of those have tiny pockets that most likely won't help with heating hands. For example, leather pants have only 2 storage, meaning you can't comfortably fit your hands in there.
6 would be a good minimum storage level to start granting POCKETS tag.

Also, leg-less armor generally shouldn't have pockets. Keeping your hands inside chest rig's pockets would be hard and uncomfortable. It would be easier to just stick them inside your pants (even pocket-less ones) and we don't simulate that.
Exceptions are clothes like hoodies that have their pockets obviously on the proper level.

**Comment by Coolthulhu** [comment]
> Add COLLAR flag to survivor duster and sleeveless survivor duster.

Regular duster doesn't have a collar.
Collar doesn't make much sense with duster's 10 warmth.
It already has a hood that it doesn't need...

**Comment by Darkyhard** [comment]
> Why fancy for SMG briefcase? And 18 storage is a bit too much.

I think it should be fancy because it is designed to look like an average suitcase/briefcase, which is fancy.

As for storage, I've used briefcase's storage amount as a base, and I don't think that gun parts in the suitcase will occupy more than 6 units of storage.

> Some of those have tiny pockets that most likely won't help with heating hands. For example, leather pants have only 2 storage, meaning you can't comfortably fit your hands in there.
6 would be a good minimum storage level to start granting POCKETS tag.

So, which ones shouldn't have POCKETS tag? It's hard for me to tell just by description and amount of storage space.

> Keeping your hands inside chest rig's pockets would be hard and uncomfortable. It would be easier to just stick them inside your pants (even pocket-less ones) and we don't simulate that.

If you're trying to keep your hands from being frostbitten, you won't mind if it is comfortable to hold your hands in pockets or not.

> Regular duster doesn't have a collar.
Collar doesn't make much sense with duster's 10 warmth.
It already has a hood that it doesn't need...

Normal trenchcoats don't have a collar or a hood, yet survivor trenchcoats have both. I think that dusters could work in a similar way.

**Comment by Coolthulhu** [comment]
> So, which ones shouldn't have POCKETS tag? It's hard for me to tell just by description and amount of storage space.

As I said, 6 is a good storage space minimum for pants.
4 pockets, 2 in the back (generally too small to fit a hand), 2 in the front. This means that the back pockets would be 1 volume, front ones 2. 2 volume pocket would be a good minimum  to hold a hand in comfortably.

> If you're trying to keep your hands from being frostbitten, you won't mind if it is comfortable to hold your hands in pockets or not.

As I said, this logic doesn't work here, because if you were keeping your hands from being frostbitten, you'd keep them inside sleeves, inside pants (not just pockets), up the shirt etc.
All of those would be more effective than keeping them in pockets too small to fit a smartphone.

Since we don't allow keeping hands under the shirt, we shouldn't allow the less effective and less comfortable things like keeping them in chest rig pockets.
Plus, adding POCKETS tag to chest rig doesn't actually make any sense from gameplay perspective - the survivor will never use that, because chest rig has 0 warmth. It would only result in a misleading message in item description, saying that you can use them to warm your hands while you actually can't.

> Normal trenchcoats don't have a collar or a hood, yet survivor trenchcoats have both. I think that dusters could work in a similar way.

OK, I missed the trenchcoats.
Though honestly it would be better to remove hoods and collars from survivor trenchcoats, because they don't make much sense with 10 warmth (the most significant result of that is that the description block for them is huge), but I guess wool/fur ones could use that.

**Comment by Darkyhard** [comment]
> 4 pockets, 2 in the back (generally too small to fit a hand), 2 in the front. This means that the back pockets would be 1 volume, front ones 2. 2 volume pocket would be a good minimum to hold a hand in comfortably.

Why not 4 units of storage then? A pair of pants may have two 2-volume front pockets, but no back pockets, in which case you would still be able to hold a hand in them comfortably.

> As I said, this logic doesn't work here, because if you were keeping your hands from being frostbitten, you'd keep them inside sleeves, inside pants (not just pockets), up the shirt etc.
All of those would be more effective than keeping them in pockets too small to fit a smartphone.
Since we don't allow keeping hands under the shirt, we shouldn't allow the less effective and less comfortable things like keeping them in chest rig pockets.

Got it.

By the way, would those things make it into the game if someone managed to find a good way to handle it?

**Comment by Coolthulhu** [comment]
> Why not 4 units of storage then? A pair of pants may have two 2-volume front pockets, but no back pockets, in which case you would still be able to hold a hand in them comfortably.

It could happen, but I'd keep it at 6 just to make the whole pocket thing mean something.
Not much, because it only applies pants' warmth (which generally isn't much), but still.
Fur pants could be an exception, because they're specifically made for warmth and not storage.

> However, I thought that you could still use item's pockets to protect your hands from wind even if it has 0 warmth.

Currently wind ignores pockets, hoods and collars. This sometimes causes a message spam when wind cools down your hands, but pockets warm them up back to comfortable.
Still, I'd say that pockets made just for storage probably wouldn't be good at protecting from wind.
> By the way, would those things make it into the game if someone managed to find a good way to handle it?

It would be cool to have some early game frostbite prevention mechanic, but for it to be done well, it would need to be something different than "you no longer get frostbite because you can always warm hands up".
Something like restricting it to resting, easy crafting, maybe reading. Maybe reading only when there's a seat+table nearby?

**Comment by Zireael07** [comment]
What has the "seat + table for reading" idea do with frostbite?

And on topic, why does wind ignore pockets? It'd fix the spam...

**Comment by Coolthulhu** [comment]
> What has the "seat + table for reading" idea do with frostbite?

The part where putting hands inside your clothing that isn't designed for hands (say, up a shirt or down pocket-less pants) means you can't use them to hold the book.

> And on topic, why does wind ignore pockets?

Because it doesn't check for it.
As in, no one implemented wind+hood/pocket/collar interaction.

**Comment by chaosvolt** [comment]
Hmm. Interesting ideas overall.

Only thing I can say is that when I made the briefcase wearable, the only reason I didn't give it volume was due to possible Fun that might occur if you decide to wield it, especially if the added volume is keeping you from dumping some items. XP

**Comment by Cyrano7** [comment]
http://imgur.com/wBm6tBQ :D 
@chaosvolt But yea that's a good reason for the suitcase smg not to have volume.

**Comment by illi-kun** [comment]
Cyrano7 is right, I see no reason to add storage value to suitcase SMG.

**Comment by chaosvolt** [comment]
True. That's mostly a gameplay reason rather than a realism reason. XP

**Comment by kevingranade** [comment]
The realism reason would be, "the suitcase is full of SMG". A
self-contained SMG fitting in a briefcase with space left over doesn't
necessarily imply a SMG built into a briefcase is also going to have space
left over. The feed mechanism might be different, it needs to be oriented a
particular way, it needs to be fixed in place in the briefcase, it might
need to catch shell casings.

**Comment by chaosvolt** [comment]
True. Basically the question is how much scrutiny the item is supposed to hold up to.

Neither of the most common types I know of leave much storage space, let alone enough to pass as a real briefcase. There's the classic "MP5 built into a proper briefcase" kind, then there's the "foldable weapon that vaguely resembles a briefcase when folded" which is what I modeled the sprite after.

**Comment by chaosvolt** [comment]
So in other words, I went with gameplay reasons for not giving it storage without thinking of the realism argument, but even the realism argument doesn't necessarily favor giving it storage. o3o

File: data/json/items/armor.json [hunk 10]
```diff
@@ -3344,7 +3344,7 @@
         "phase" : "solid",
         "encumbrance" : 0,
         "bashing" : -5,
-        "flags" : ["VARSIZE"],
+        "flags" : ["VARSIZE", "POCKETS"],
```
Comment (Coolthulhu) [inline data/json/items/armor.json]: Those pockets aren't really usable to warm hands. Even if they were - 5 warmth

File: data/json/items/armor.json [hunk 11]
```diff
@@ -4351,7 +4351,7 @@
         "phase" : "solid",
         "encumbrance" : 15,
         "bashing" : 6,
-        "flags" : ["VARSIZE", "STURDY", "OUTER"],
+        "flags" : ["VARSIZE", "POCKETS", "STURDY", "OUTER"],
```
Comment (Coolthulhu) [inline data/json/items/armor.json]: The pockets are most likely unsuited to holding hands in them - they're made for armor plates and thus unlikely to provide the warmth the vest itself provides and also located in awkward locations.
Same for all the variants with armor inserted.

File: data/json/items/armor.json [hunk 12]
```diff
@@ -1005,7 +1005,7 @@
         "phase" : "solid",
         "encumbrance" : 5,
         "bashing" : -4,
-        "flags" : ["VARSIZE"],
+        "flags" : ["VARSIZE", "POCKETS"],
```
Comment (Coolthulhu) [inline data/json/items/armor.json]: With 5 warmth and 4 storage, it isn't really suited to have the POCKETS flag. Same for the two variants below.

**Comment by Darkyhard** [comment]
Done,

**Pull request merged**